compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1434,9 +1434,20 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         span: Span,
         did: DefId,
         item_segment: &hir::PathSegment<'_>,
+        is_type_alias: bool,
```

Comment:
nit: we can re-deduce this with `def_kind(did)` -- not sure if it's worth, but adding a new param is kinda noisy.